src/ddmd/expression.d

```diff
@@ -15410,6 +15411,24 @@ extern (C++) final class CmpExp : BinExp
             if ((t1.ty == Tarray || t1.ty == Tsarray) && (t2.ty == Tarray || t2.ty == Tsarray))
             {
                 semanticTypeInfo(sc, t1.nextOf());
+
+                // Lowering to template call
```

Comment:
Would be nice to document what the template call is (it's not obvious from the sequence of AST node creations and links).

<ihaveadream>
Would be great to have the ability to create an AST fragment from a string source, a la:

```d
auto node = makeExpression!"__cmp($1, $2)"(e1, e2);
```

whereby `makeExpression` parses the string during compilation (of dmd), creates the appropriate AST fragment, places the two expressions in lieu of `$1` and `$2`, and returns the created expression node.

I assume a large hunk of compiler code would be drastically simplified.
</ihaveadream>